Core/src/org/sleuthkit/autopsy/modules/leappanalyzers/LeappFileProcessor.java

```diff
@@ -437,29 +557,66 @@ private void getArtifactNode(Document xmlinput) {
 
     }
 
+    private String getXmlFileIdentifier(String fileName) {
+        return String.format("file: %s, filename: %s",
+                this.xmlFile == null ? "<null>" : this.xmlFile,
+                fileName == null ? "<null>" : fileName);
+    }
+
+    private String getXmlAttrIdentifier(String fileName, String attributeName) {
+        return String.format("attribute: %s %s",
+                attributeName == null ? "<null>" : attributeName,
+                getXmlFileIdentifier(fileName));
+    }
+
     private void getAttributeNodes(Document xmlinput) {
 
         NodeList attributeNlist = xmlinput.getElementsByTagName("AttributeName"); //NON-NLS
         for (int k = 0; k < attributeNlist.getLength(); k++) {
-            List<String> attributeList = new ArrayList<>();
             NamedNodeMap nnm = attributeNlist.item(k).getAttributes();
             String attributeName = nnm.getNamedItem("attributename").getNodeValue();
+
             if (!attributeName.toLowerCase().matches("null")) {
                 String columnName = nnm.getNamedItem("columnName").getNodeValue();
                 String required = nnm.getNamedItem("required").getNodeValue();
                 String parentName = attributeNlist.item(k).getParentNode().getParentNode().getAttributes().getNamedItem("filename").getNodeValue();
 
-                attributeList.add(attributeName.toLowerCase());
-                attributeList.add(columnName.toLowerCase());
-                attributeList.add(required.toLowerCase());
+                BlackboardAttribute.Type foundAttrType = null;
+                try {
+                    foundAttrType = Case.getCurrentCase().getSleuthkitCase().getAttributeType(attributeName.toUpperCase());
+                } catch (TskCoreException ex) {
+                    logger.log(Level.SEVERE, String.format("There was an issue that arose while trying to fetch attribute type for %s.", attributeName), ex);
+                }
+
+                if (foundAttrType == null) {
+                    logger.log(Level.SEVERE, String.format("No known attribute mapping found for [%s]", getXmlAttrIdentifier(parentName, attributeName)));
+                }
+
+                if (required != null && required.compareToIgnoreCase("yes") != 0 && required.compareToIgnoreCase("no") != 0) {
+                    logger.log(Level.SEVERE, String.format("Required value %s did not match 'yes' or 'no' for [%s]",
+                            required, getXmlAttrIdentifier(parentName, attributeName)));
+                }
+
+                if (columnName == null) {
+                    logger.log(Level.SEVERE, String.format("No column name provided for [%s]", getXmlAttrIdentifier(parentName, attributeName)));
```

Comment:
I'm not sure these 3 log messages are severe. I think a warning would be better.

Reply:
If this flag gets hit, I think that identifies malformed xml.

Reply:
Okay, sounds good.